internal/storage/index/index.go

```diff
@@ -447,6 +449,33 @@ func (idx *index) Inspect() map[string]meta {
 	return entries
 }
 
+func (idx *index) InspectPolicies(ctx context.Context) (map[string]*responsev1.InspectPoliciesResponse_Result, error) {
+	policyIDs, err := idx.ListPolicyIDs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list policies: %w", err)
+	}
+	if len(policyIDs) == 0 {
+		return nil, nil
+	}
+
+	policies, err := idx.LoadPolicy(ctx, policyIDs...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load policies: %w", err)
+	}
+
+	results := make(map[string]*responsev1.InspectPoliciesResponse_Result)
+	for _, p := range policies {
+		actions := policy.ListActions(p.Policy)
+		if len(actions) > 0 {
+			results[p.FQN] = &responsev1.InspectPoliciesResponse_Result{
+				Actions: actions,
+			}
+		}
+	}
```

Comment:
Instead of loading all the policies into memory at once and iterating through them, it'd be better to load them one-by-one.